app/routes.py

```diff
@@ -1,2 +1,274 @@
+    if not task:
+        return make_response(f"Task #{task_id} Not Found", 404)
+
+    if request.method == "GET":
+        if not task.completed_at:
+            is_complete = False
+        else:
+            is_complete = True
+        task_response = {"task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": is_complete,
+        }}
+        if task.goal_id:
+            task_response["task"]["goal_id"] = task.goal_id
+        return make_response(task_response, 200)
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        if ("completed_at" not in form_data or
+                form_data["completed_at"] == None):
+            is_complete = False
+        else:
+            is_complete = True
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+        db.session.commit()
+        task_response = {"task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": is_complete
+        }}
+        return make_response(task_response, 200)
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        task_response = {
+            "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
+        }
+        return make_response(task_response, 200)
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def handle_task_complete(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        return make_response(f"Task #{task_id} Not Found", 404)
+
+    task.completed_at = datetime.datetime.utcnow()
+    db.session.commit()
+    channel = "channel=task-notifications"
+    text = f"text=Someone just completed the task {task.title}"
+
+    path = f"https://slack.com/api/chat.postMessage?{channel}&{text}"
+
+    SLACKBOT_AUTH = os.environ.get(
+        "SLACKBOT_AUTH")
+
+    requests.post(path, headers={'Authorization': f'Bearer {SLACKBOT_AUTH}'})
+
+    task_response = {"task": {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": True
+    }}
+    return make_response(task_response, 200)
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def handle_task_incomplete(task_id):
+    # get current data, set the thing
+    task = Task.query.get(task_id)
+    if not task:
+        return make_response(f"Task #{task_id} Not Found", 404)
+
+    task.completed_at = None
+    db.session.commit()
+
+    task_response = {"task": {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": False
+    }}
+    return make_response(task_response, 200)
+
+
+@goals_bp.route("/", methods=["GET", "POST"])
+def handle_goals():
```

Comment:
Similar feedback about splitting these functions, and moving validation and dictionary-handling logic around that I made for `Task` will also apply to `Goals`. These are common themes for any model with CRUD operations.